include/util/coordinate_calculation.hpp

```diff
@@ -95,6 +95,23 @@ inline std::pair<double, FloatCoordinate> projectPointOnSegment(const FloatCoord
             }};
 }
 
+// find the closest distance between a coordinate and a segment
+// O(1)
+double findClosestDistance(const Coordinate coordinate,
+                           const Coordinate segment_begin,
+                           const Coordinate segment_end);
+
+// find the closest distance between a coordinate and a set of coordinates
+// O(|coordinates|)
+double findClosestDistance(const Coordinate coordinate, const std::vector<Coordinate> &coordinates);
+
+// find the closes distance between two sets of coordinates
+// O(|lhs| * |rhs|)
+double findClosestDistance(const std::vector<Coordinate> &lhs, const std::vector<Coordinate> &rhs);
+
+// checks if two sets of coordinates describe a parallel set of ways
+bool areParallel(const std::vector<Coordinate> &lhs, const std::vector<Coordinate> &rhs);
```

Comment:
Hm these interfaces are maybe a bit to strict - do we want to hardcode vector<Coord> here, or could we use Iterators? (I just want to avoid situations we already have in the code base where we create vectors, call a function, and then throw them away only)